src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMSource.java

```diff
@@ -440,13 +440,15 @@ protected final void retrieve(@CheckForNull SCMSourceCriteria criteria,
             String fullName = repoOwner + "/" + repository;
             final GHRepository repo = github.getRepository(fullName);
             listener.getLogger().format("Looking up %s%n", HyperlinkNote.encodeTo(repo.getHtmlUrl().toString(), fullName));
-            try {
-                repositoryUrl = repo.getHtmlUrl();
-                collaboratorNames = new HashSet<>(repo.getCollaboratorNames());
-            } catch (FileNotFoundException e) {
-                // not permitted
-                listener.getLogger().println("Not permitted to query list of collaborators, assuming none");
-                collaboratorNames = Collections.emptySet();
+            if (!repo.isPrivate()) { // random people cannot create pull requests to private repositories
```

Comment:
@stephenc thanks, now it's clear to me.

What about adding the checkbox (which will be turned on by default) that enables the behavior you described and ignore it when disabled? If you agree that checkbox makes sense I'll try to pull up my java coding skills to the level that I will be able to come up with such PR =)